compiler/noirc_frontend/src/elaborator/patterns.rs

```diff
@@ -386,13 +388,8 @@ impl<'context> Elaborator<'context> {
         // This check is necessary to maintain the same definition ids in the interner. Currently, each function uses a new resolver that has its own ScopeForest and thus global scope.
         // We must first check whether an existing definition ID has been inserted as otherwise there will be multiple definitions for the same global statement.
         // This leads to an error in evaluation where the wrong definition ID is selected when evaluating a statement using the global. The check below prevents this error.
-        let mut global_id = None;
-        let global = self.interner.get_all_globals();
-        for global_info in global {
-            if global_info.local_id == self.local_module && global_info.ident == name {
-                global_id = Some(global_info.id);
-            }
-        }
+        let global_id =
+            self.interner.find_global_by_local_and_ident(self.local_module, name.clone());
```

Comment:
So here we could unpack the `definition` (or change the parameter to contain the `GlobalId` itself, which the caller already checked), and look up by ID. Would that make that redundant?

Reply:
I added an extra assertion in https://github.com/noir-lang/noir/pull/6282/commits/ce3fa081e16ef7888f02e269d1a0951a6b410674 that the result from this `find` matches the input `definition`, so it seems reasonable to me that we should look up by ID and probably don't need the local+ident based search.

If we don't find the global by local+ident, we end up pushing another item into `definitions` with the same content as already exists. If we instead looked up by the `GlobalId` contained in `definition`, we could reuse the same `definition_id`.

Reply:
Ah, just saw https://github.com/noir-lang/noir/issues/5001#issuecomment-2180721837 saying we should be able to get rid of this check.